### PR TITLE
fix: fix copy modal

### PR DESCRIPTION
**Pull request by pyphilia**
Enable back copy button, using the new item selection modal.

### CODE REVIEW

File: src/components/collection/copyModal/RootNavigationTree.tsx [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+  const { data: recentItems, isLoading } = hooks.useAccessibleItems(
+    // you can move into an item you have at least write permission
+    {
+      permissions: [PermissionLevel.Admin, PermissionLevel.Write],
+      types: [ItemType.FOLDER],
+    },
+    { pageSize: 5 },
+  );
+  const recentFolders = recentItems?.data?.filter(
+    ({ type }) => type === ItemType.FOLDER,
```
Comment (LinaYahya) [inline src/components/collection/copyModal/RootNavigationTree.tsx:40]: As recent Items should be a folder type, Why are we filtering it again?

File: cypress/support/server.ts [hunk 2]
```diff
@@ -70,29 +69,30 @@ export const redirectionReply = {
...
-          creator?.id === currentMember.id && !path.includes('.'),
-      );
-      return reply(own);
+    ({ url, reply }) => {
+      const params = new URL(url).searchParams;
+
+      const page = parseInt(params.get('page') ?? '1', 10);
+      const pageSize = parseInt(params.get('pageSize') ?? '10', 10);
+
+      // as { page: number; pageSize: number };
```
Comment (spaenleh) [inline cypress/support/server.ts:84]: remove ?

File: cypress/support/server.ts [hunk 3]
```diff
@@ -70,29 +69,30 @@ export const redirectionReply = {
...
+
+      // warning: we don't check memberships
+      const root = items.filter((i) => !i.path.includes('.'));
+
+      // todo: filter
```
Comment (spaenleh) [inline cypress/support/server.ts:89]: Is this todo still relevant ?

File: package.json [hunk 4]
```diff
@@ -36,7 +36,7 @@
     "@graasp/query-client": "3.4.1",
     "@graasp/sdk": "4.4.0",
     "@graasp/translations": "1.25.3",
-    "@graasp/ui": "4.15.0",
+    "@graasp/ui": "github:graasp/graasp-ui#row-menu",
```
Comment (spaenleh) [inline package.json:39]: Don't forget to change

File: src/components/collection/CopyButton.tsx [hunk 5]
```diff
@@ -36,29 +32,31 @@ export const useCopyAction = (id?: string) => {
...
   // todo: set notifier for copy
-  const copy = ({ to }: { to: string }) => {
+  const copy = (to: string | undefined) => {
     // remove loading icon on callback
     // do not set parent if it is root
     const payload: Parameters<typeof copyItems>[0] = {
       ids: [id],
     };
 
     // if the location to copy the item is MyItems or SharedItems root, then set the payload.to argument to be undefined
```
Comment (spaenleh) [inline src/components/collection/CopyButton.tsx:42]: remove comment ?

File: src/components/collection/copyModal/AccessibleNavigationTree.tsx [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+
+const PAGE_SIZE = 10;
+
+const AccessibleNavigationTree = ({
+  onClick,
+  onNavigate,
+  selectedId,
+}: AccessibleNavigationTreeProps): JSX.Element => {
+  const [page, setPage] = useState(1);
+  // todo: show only items with admin rights
```
Comment (spaenleh) [inline src/components/collection/copyModal/AccessibleNavigationTree.tsx:24]: Is this still relevant ? We filter on write and admin as I see.

File: src/components/collection/copyModal/AccessibleNavigationTree.tsx [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+    return (
+      <>
+        <Skeleton height={50} />
+        <Skeleton height={50} />
+        <Skeleton height={50} />
+      </>
+    );
+  }
+
+  return <Alert severity="error">An unexpected error happened</Alert>;
```
Comment (spaenleh) [inline src/components/collection/copyModal/AccessibleNavigationTree.tsx:62]: Translate ?

File: src/components/collection/copyModal/ChildrenNavigationTree.tsx [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+  onNavigate,
+  isDisabled,
+}: ChildrenNavigationTreeProps): JSX.Element => {
+  const { t } = useLibraryTranslation();
+  const { hooks } = useContext(QueryClientContext);
+  const { data: children, isLoading } = hooks.useChildren(
+    selectedNavigationItem.id,
+  );
+  // TODO: use hook's filter when available
+  const folders = children?.filter((f) => f.type === ItemType.FOLDER);
```
Comment (spaenleh) [inline src/components/collection/copyModal/ChildrenNavigationTree.tsx:33]: I think the filter is available now no ? 
```suggestion
  const { data: children, isLoading } = hooks.useChildren(
    selectedNavigationItem.id,
    { types: [ItemType.FOLDER] }
  );
```

File: src/components/collection/copyModal/ChildrenNavigationTree.tsx [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+  if (isLoading) {
+    return (
+      <>
+        <Skeleton height={50} />
+        <Skeleton height={50} />
+        <Skeleton height={50} />
+      </>
+    );
+  }
+  return <Alert severity="error">An unexpected error happened</Alert>;
```
Comment (spaenleh) [inline src/components/collection/copyModal/ChildrenNavigationTree.tsx:60]: Translate.

File: src/components/collection/copyModal/RootNavigationTree.tsx (hunk 1 above)
Comment (spaenleh) [reply src/components/collection/copyModal/RootNavigationTree.tsx:40]: I agree, I think this filter can be removed.

File: src/components/collection/copyModal/RootNavigationTree.tsx [hunk 10]
```diff
@@ -0,0 +1,87 @@
...
+    return (
+      <>
+        <Skeleton height={50} />
+        <Skeleton height={50} />
+        <Skeleton height={50} />
+      </>
+    );
+  }
+
+  return <Alert severity="error">An unexpected error happened</Alert>;
```
Comment (spaenleh) [inline src/components/collection/copyModal/RootNavigationTree.tsx:84]: Translate

**Review by spaenleh — CHANGES_REQUESTED** [review]
Looks good thank you for the changes, just a few tweaks and we are good to go !

**Review by spaenleh — APPROVED** [review]

**Pull request merged**